File: LeetCode/Python Solutions/Pass the Pillow/pass.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        cycle_length = 2 * (n - 1)
        effective_time = time % cycle_length

        if effective_time < n:
            return effective_time + 1
        else:
            return 2 * n - 1 - effective_time

File: LeetCode/Java Solutions/Pass the Pillow/pass.java
class Solution {
    public int passThePillow(int n, int time) {
        int cycleLength = 2 * (n - 1);
        int effectiveTime = time % cycleLength;

        if (effectiveTime < n) {
            return effectiveTime + 1;
        } else {
            return 2 * n - 1 - effectiveTime;
        }
    }
}
